src/DropdownItem.js

```diff
@@ -40,7 +43,9 @@ class DropdownItem extends React.Component {
       this.props.onClick(e);
     }
 
-    this.context.toggle();
+    if (this.props.toggle) {
+      this.context.toggle();
```

Comment:
Rethinking this, bool seems fine for preventing toggle. But, if a function is passed to `toggle` prop, we should probably call that instead of context. That can be an enhancement pr that should be backwards compatible.